include/cuco/detail/static_set/static_set.inl

```diff
@@ -323,47 +325,68 @@ static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::siz
 }
 
 template <class Key,
-          class Extent,
           cuda::thread_scope Scope,
+          class Extent,
           class KeyEqual,
           class ProbingScheme,
           class Allocator,
           class Storage>
 constexpr auto
-static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::capacity()
+static_set<Key, Scope, Extent, KeyEqual, ProbingScheme, Allocator, Storage>::capacity()
   const noexcept
 {
   return storage_.capacity();
 }
 
 template <class Key,
-          class Extent,
           cuda::thread_scope Scope,
+          class Extent,
           class KeyEqual,
           class ProbingScheme,
           class Allocator,
           class Storage>
-constexpr static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::key_type
-static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::empty_key_sentinel()
+constexpr static_set<Key, Scope, Extent, KeyEqual, ProbingScheme, Allocator, Storage>::key_type
+static_set<Key, Scope, Extent, KeyEqual, ProbingScheme, Allocator, Storage>::empty_key_sentinel()
   const noexcept
 {
   return empty_key_sentinel_;
 }
 
 template <class Key,
-          class Extent,
           cuda::thread_scope Scope,
+          class Extent,
           class KeyEqual,
           class ProbingScheme,
           class Allocator,
           class Storage>
 template <typename... Operators>
-auto static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::ref(
+auto static_set<Key, Scope, Extent, KeyEqual, ProbingScheme, Allocator, Storage>::ref(
   Operators...) const noexcept
 {
   static_assert(sizeof...(Operators), "No operators specified");
   return ref_type<Operators...>{
     cuco::empty_key<key_type>(empty_key_sentinel_), predicate_, probing_scheme_, storage_.ref()};
 }
+
+template <class Key, cuda::thread_scope Scope, class... Args>
+constexpr auto make_static_set(Args&&... args)
+{
+  // TODO don't repeat defaults
```

Comment:
Yeah, that's also unelegant. Currently, we need to mirror every default tparam from `static_set` explicitly. If we change the default in `static_set`, but forget to change it in the factory, that would be bad.
Maybe I can instantiate a set with default tparams first, and then extract the tparams from its member type aliases:
```cpp
using default_set = static_set<Key>;
using default_allocator = typename default_set::allocator_type;
...
```